pythonforandroid/python.py

```diff
@@ -197,6 +197,7 @@ def add_flags(include_flags, link_dirs, link_libs):
             add_flags(' -I' + ' -I'.join(recipe.get_include_dirs(arch)),
                       ' -L' + join(recipe.get_build_dir(arch.arch), '.libs'),
                       ' -lffi')
+            env["LDFLAGS"] += ' -L. -fuse-ld=lld'
```

Comment:
I would definitely make this part optional since it seems to work e.g. on Ubuntu without that line, right?
So how about something like:
```python
# using the lld linker seems optional on some systems e.g. Ubuntu 18.04,
# but mandatory on others e.g. Fedora 29, WSL...
# if you get undefined references to ffi, consider installing lld
exitcode, _ = subprocess.getstatusoutput('which lld')
if exitcode == 0:
    env["LDFLAGS"] += ' -L. -fuse-ld=lld'
else:
    logger.warning('lld not found, linking without it')
```
This is an example, but you can actually give a more comprehensive warning message to point people to the correct direction.
I'm suggesting to make it optional, because otherwise it would by default break on all systems that don't have lld already installed. And the error is not explicit unless you look in the `config.log`

Reply:
I totally agree with @AndreMiras...and I may add that you should  make the patch `remove-fix-cortex-a8.patch` conditional, because if some system don't use the `lld`, then we don't want that `patch` applied.

Also I feel that these `linker flags` should go in `get_recipe_env`, not in here...but maybe I'm wrong...@Aralox, why this `ldflags` should be conditional to libffi build?

Reply:
Thanks for the review opacam. I've made the patch conditional. I'll push my code, but will run tests overnight and get back to this later in the week.
And sure, I'll move the linker flags to get_recipe_env(). I put it specifically under libffi just to isolate my changes, but it should be fine to apply it to the whole python build as it uses clang.